src/Microsoft.AspNetCore.Sockets.Client.Http/WebSocketsTransport.cs

```diff
@@ -178,10 +178,31 @@ public WebSocketsTransport(HttpOptions httpOptions, ILoggerFactory loggerFactory
         {
             try
             {
+                // Skip the 0 byte read at first because we will receive the Hub handshake response shortly
+                var endOfMessage = false;
                 while (true)
                 {
-                    var memory = _application.Output.GetMemory();
+#if NETCOREAPP2_1
+                    // If there was a read that had a 'false' EndOfMessage then we should skip the 0 byte read since there is an in-progress frame
```

Comment:
Having a previous `EndOfMessage = false` frame doesn't guarantee that there's data available to be read from the socket though. The server may not have sent the next fragment yet. Is it worth saving a few `0`-byte reads?